### ISSUE TITLE
Sidekiq exception handlers now take three arguments (DEPRECATION since sidekiq 7.1.5, will become error in 8)

### ISSUE THREAD

**Issue by timdiggins** [bugsnag/bugsnag-ruby#804]
### Describe the bug

Sidekiq has (since 7.1.5) changed the api for its exception handlers.
https://github.com/sidekiq/sidekiq/blob/main/Changes.md#715

### Steps to reproduce

Use a version of sidekiq since 7.1.5 (e.g. 7.2.0 which is latest as of date of writing).
Execute a Sidekiq job with an error like

```
class ErroringJob
  include Sidekiq::Job
  def perform
     raise "Deliberate error"
  end
end

ErroringJob.perform_async
```

it will output

```
DEPRECATION: Sidekiq exception handlers now take three arguments, see #<Proc:0x00007fad0325ccd8 /pathto/ruby/3.0.6/lib/ruby/gems/3.0.0/gems/bugsnag-6.26.0/lib/bugsnag/integrations/sidekiq.rb:53>
```

### Environment

* Bugsnag version: 6.26.0
* Ruby version: 3.0.6
* Bundle version: 2.2.33
* Integration framework version:
    * Sidekiq: 7.2.0

**Comment by clr182** [comment]
Hi @timdiggins 
Thanks for reaching out. We are aware of this deprecation warning. We have an item on our backlog aimed at fixing this warning. While we don't have an ETA for the release of this functionality, you can rest assured that a fix will be in place before the warning becomes an error in v8 of Sidekiq.

**Comment by stevenharman** [comment]
This was addressed in #796 and released in 6.26.1.

**Comment by pic** [comment]
As far as I can tell, the deprecation warning is still displayed in 6.26.4.
The handler is defined with a default value:

```
server.error_handlers << proc do |ex, _context, _config = nil|
```

and in sidekiq check this still has arity == 2

```
        if handler.arity == 2
          # TODO Remove in 8.0
          logger.info { "DEPRECATION: Sidekiq exception handlers now take three arguments, see #{handler}" }
```

**Comment by mclack** [comment]
Hi @pic 

Apologies for the delay in response.

Can you please confirm whether you are still seeing this warning being displayed when using a `bugsnag-ruby` version of 6.26.1 or above? With the change [here](https://github.com/bugsnag/bugsnag-ruby/blob/7bf79149119f5b685d55ebfe12ee96f6e909749c/lib/bugsnag/integrations/sidekiq.rb#L53), we would expect this warning to no longer be logged based on the condition of `handler.arity == 2`, as the arity of the handler should now be `-3`.

Please let us know if you are still seeing this deprecation warning. It would be useful to see the version of `bugsnag-ruby` you are using, as well as any warnings or errors being displayed.

**Comment by pic** [comment]
Hi @mclack 

We are using 6.27. As far as I can tell you this declaration:

 `server.error_handlers << proc do |ex, _context, _config = nil`
 
is considered of arity 2, I guess because of the default value.

For comparison, I've just noticed how newrelic handles this:

```
        # Sidekiq 3.0.0 - 7.1.4 expect error_handlers to have 2 arguments
        # Sidekiq 7.1.5+ expect error_handlers to have 3 arguments
        config.error_handlers << proc do |error, _ctx, *_|
          NewRelic::Agent.notice_error(error)
        end
```

Thx

**Comment by matthewjhowells** [comment]
Hi @pic, 
This should have been fixed in bugsnag-ruby version [6.26.1](https://github.com/bugsnag/bugsnag-ruby/releases/tag/v6.26.1) so we wouldn’t expect you to be seeing this depreciation warning any more. Could you please confirm which version of Bugsnag-ruby you are using? If you are running version 6.26.1 or higher, could you please share a reproduction with us to investigate further?

If the reproduction might contain anything sensitive, please feel free to share it with us at [support@bugsnag.com](mailto:support@bugsnag.com) instead.

**Comment by pic** [comment]
```
$ bundle show bugsnag
[...]rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bugsnag-6.27.1
```

Still displaying the deprecation warning.
I don't have resources right now to try that, but I think that a new RoR 7.0.x application, with a worker, that is raising an exception in `perform` would expose this.
